pkg/operator/ceph/cluster/osd/pod.go

```diff
@@ -90,37 +91,49 @@ func (c *Cluster) makeDeployment(nodeName string, devices []rookalpha.Device, se
 		logger.Infof("no data dir provided")
 	}
 
-	// Mount the path to the directory-based osd unless it is already available under the dataDirHostPath
-	if osd.IsDirectory && !strings.HasPrefix(osd.DataPath, k8sutil.DataDir) {
-		volumeName := k8sutil.PathToVolumeName(osd.DataPath)
-		dataDirSource := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: osd.DataPath}}
-		volumes = append(volumes, v1.Volume{Name: volumeName, VolumeSource: dataDirSource})
-		volumeMounts = append(volumeMounts, v1.VolumeMount{Name: volumeName, MountPath: osd.DataPath})
-	}
-
-	// Mount the required device
-	if !osd.IsDirectory {
+	// Mount the path to the directory-based osd
```

Comment:
this comment should be inside the block for `if osd.IsDirectory` since it's only applicable in that scope